### PR TITLE
[VL] Fix Filter node falling back to Spark whenever there is Timestamp column in input (child node)

### PR DESCRIPTION
**Pull request by mskapilks**
## What changes were proposed in this pull request?
Currently if the input to filter node contains any timestamp column, it fallbacks to Spark even though filter predicate may not be on timestamp column. Current check is very strict.

(Fixes: \#ISSUE-ID)

## How was this patch tested?
Unit tests in PR

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: cpp/velox/substrait/SubstraitToVeloxPlanValidator.cc [hunk 1]
```diff
@@ -829,12 +830,6 @@ bool SubstraitToVeloxPlanValidator::validate(const ::substrait::FilterRel& filte
     logValidateMsg("native validation failed due to: Validation failed for input types in FilterRel.");
     return false;
   }
-  for (const auto& type : types) {
```
Comment (mskapilks) [inline cpp/velox/substrait/SubstraitToVeloxPlanValidator.cc:832]: Removed this completely to check if all UTs passes, as there are specific validation for timestamp cases. If some case are missing, I can add better validation check which can validate if predicate is on timestamp column.

**Comment by mskapilks** [comment]
The one failure is not related to my change.
```
error: building krb5:x64-linux-avx failed with: BUILD_FAILED
Elapsed time to handle krb5:x64-linux-avx: 1.5 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
```

**Comment by rui-mo** [comment]
@mskapilks Just restart this test.

File: backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala [hunk 2]
```diff
@@ -734,6 +734,34 @@ class TestOperator extends VeloxWholeStageTransformerSuite with AdaptiveSparkPla
...
+      runQueryAndCompare("select c1, c2 from ts where c1 = 1") {
+        checkOperatorMatch[FileSourceScanExecTransformer]
+      }
+
+      // Fallback should only happen when there is a filter on timestamp column
+      runQueryAndCompare(
+        "select c1, c2 from ts where" +
+          " c2 = timestamp'1965-01-01 10:11:12.123456'") {
+        df: DataFrame =>
+          assert(df.queryExecution.executedPlan.find(_.isInstanceOf[FileSourceScanExec]).isDefined)
```
Comment (rui-mo) [inline backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala:753]: I feel we don't need to check a fallback plan as native execution is preferred. To make sure the query is runnable with `runQueryAndCompare` is enough. How do you think?

**Review by rui-mo — COMMENTED** [review]
Thanks. It makes sense to apply more specific fallback logic on timestamp.

File: backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala (hunk 2 above)
Comment (mskapilks) [reply backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala:753]: Make sense

**Comment by mskapilks** [comment]
The current failure seems due to some package download failure

**Comment by zhli1142015** [comment]
Retriggered the tests.

**Review by rui-mo — APPROVED** [review]

**Review by zhli1142015 — APPROVED** [review]

**Pull request merged**